### ISSUE TITLE
[hts-nim] error in bam.queri

### ISSUE THREAD

**Issue by hdashnow** [brentp/mosdepth#44]
Hi @brentp 

I'm getting the error message:
`[hts-nim] error in bam.queri`
multiple times, 51 in fact, last run. However, the program completes and the output looks sensible. 

The error is only occurring when running in multithreaded mode. 

The command I ran was:
`mosdepth -n -t 8 -f Homo_sapiens_assembly38.fasta  sample  sample.cram`

**Comment by brentp** [comment]
can you share a part of your cram that will recreate the error? it could be just requesting a chromosome for which there is no data. I should silence that warning if that is the case.

**Comment by brentp** [comment]
I have recently seen this problem on a network drive when, I assume, there are intermittent read failures. I suspect that's what is going on here.

**Comment by hdashnow** [comment]
I'm using a cluster with Isilon storage. If it's creating files then trying to quickly read them it could be trying before the storage syncs. Does running in multithreaded mode create intermediate files?
I'm currently trying on another cluster, but having all sorts of htslib installation issues, so I'll have to get back to you once I've got it up and running.
I can share my cram if you still think it would be helpful? https://figshare.com/s/65a9d1ee48e60da74eb1

I'm able to reproduce the bug like this:
```$ mosdepth -n -t 8 -f /group/bioi1/shared/genomes/hg38/fasta/Homo_sapiens_assembly38.fasta  D14-1564.chr22.test  D14-1564.chr22.cram
[hts-nim] error in bam.queri```
```(STR) $ mosdepth -n -f /group/bioi1/shared/genomes/hg38/fasta/Homo_sapiens_assembly38.fasta  D14-1564.chr22.test  D14-1564.chr22.cram```
(no error)

**Comment by brentp** [comment]
this must be a bug in htslib because it appears with samtools as well:
```
samtools view -@ 4 -T Homo_sapiens_assembly38.fasta sample.chr22.cram chr22 > /dev/null
```
results in `[main_samview] retrieval of region "chr22" failed due to truncated file or corrupt BAM index file`

but it runs without problem single-threaded.

**Comment by brentp** [comment]
do you mind if I share a subset of your cram (containing 1108 alignments) to open an issue in htslib?

**Comment by hdashnow** [comment]
I went it to remove sample IDs from the header before distributing the cram file more widely, so feel free to distribute.

However, today I can't replicate the error!

**Comment by brentp** [comment]
i have just opened an issue in htslib. For now the solution is to not use multiple threads.

**Comment by brentp** [comment]
this has been fixed in htslib in the version 1.9 release.

**Comment by davemcg** [comment]
FYI if you are getting a "Container header CRC32 failure" switching to samtools 1.9 also fixes the problem.
